shell/browser/native_window_mac.mm

```diff
@@ -389,7 +389,12 @@ void ViewDidMoveToSuperview(NSView* self, SEL _cmd) {
   }
 
   // Ensure we're detached from the parent window before closing.
-  RemoveChildFromParentWindow(this);
+  RemoveChildFromParentWindow();
+
+  while (!child_windows_.empty()) {
+    auto* child = child_windows_.back();
+    child->RemoveChildFromParentWindow();
```

Comment:
Should the child windows be closed when the parent is closed? See `BrowserWindow::CloseImmediately`.

Reply:
I think we should change that but as a separate PR - the previous behavior didn't close children on destroy on macOS. it does on other platforms so we _should_ should unify that.